cli/awsbatch/awsbsub.py

```diff
@@ -249,6 +242,153 @@ def _upload_and_get_command(boto3_factory, args, job_name, region, s3_bucket, lo
+    :return: composed bash command
+    """
+    command_args = shell_join(args.arguments)
+    # download awscli, if required.
+    bash_command = (
+        "curl -O https://bootstrap.pypa.io/get-pip.py >/dev/null 2>&1; "
+        "python get-pip.py --user >/dev/null 2>&1; "
+        "export PATH=~/.local/bin:$PATH >/dev/null 2>&1; "
+        "pip install awscli --upgrade --user >/dev/null 2>&1; "
+        if args.awscli
+        else ""
+    )
+    # download all job files to a tmp subfolder
+    bash_command += (
+        "mkdir -p /tmp/{FOLDER}; "
+        "aws s3 --region {REGION} sync s3://{BUCKET}/{FOLDER} /tmp/{FOLDER}; ".format(
+            REGION=region, BUCKET=s3_bucket, FOLDER=job_folder
+        )
+    )
+    if args.env:
+        # source the environment file
+        bash_command += "source /tmp/{FOLDER}{ENV_FILE}; ".format(FOLDER=job_folder, ENV_FILE=env_file)
+    # execute the job script + arguments
+    bash_command += "chmod +x /tmp/{FOLDER}{SCRIPT}; /tmp/{FOLDER}{SCRIPT} {ARGS}".format(
+        FOLDER=job_folder, SCRIPT=job_script, ARGS=command_args
+    )
+    return bash_command
+
+
+def _get_env_key_value_list(env_vars, log, env_blacklist_vars=None):
+    """
+    Get key-value environment variables list by excluding blacklisted and internal variables.
+
+    :param env_vars: list of variables to get from the environment and add to the list
+    or use 'all' to add all the list except the blacklisted ones
+    :param env_blacklist_vars: list of variable names to exclude when 'all' is passed as env_vars parameter
+    """
+    environment_blacklist = ["HOME", "LD_LIBRARY_PATH", "PATH", "PWD", "TERM", "TERMCAP", "USER", "VCPUS"]
+
+    blacklisted_vars = (
+        [var.strip() for var in env_blacklist_vars.split(",")] if env_blacklist_vars else environment_blacklist
+    )
+
+    key_value_list = []
+    for var in env_vars.split(","):
+        var_name = var.strip()
+
+        if var_name == "all":
+            # export all the env variables except the blacklisted and the internal ones
+            log.info("Environment blacklist is (%s)", blacklisted_vars)
+            for env_var in os.environ:
+                if env_var not in blacklisted_vars:
+                    _add_env_var_to_list(key_value_list, env_var, log)
+        elif var_name in os.environ:
+            # export variables explicitly specified by the user
+            _add_env_var_to_list(key_value_list, var_name, log)
+        else:
+            log.warn("Environment variable (%s) does not exist." % var_name)
+
+    return key_value_list
+
+
+def _add_env_var_to_list(key_value_list, var_name, log):
+    """
+    Get key-value environment variable and add to the given list.
+
+    Skip internal variables and functions.
+
+    :param key_value_list: list to update
+    :param var_name: var name
+    :param log: log file
+    """
+    # exclude reserved variables and functions
+    if (
+        not var_name.upper().startswith("PCLUSTER_")
```

Comment:
Small comment on why we are excluding each of these?

var_name.upper could ve called just once before if statement

else: log.warn variable discarded/excluded?